app/models/access_token.rb

```diff
@@ -0,0 +1,52 @@
+class AccessToken < ActiveRecord::Base
+  belongs_to :repository
+  attr_accessible :expiration, :token
+
+  scope :expired, ->() { where('expiration <= ?', Time.now) }
+  scope :unexpired, ->() { where('expiration > ?', Time.now) }
+
+  def self.destroy_expired
+    expired.find_each(&:destroy)
+  end
+
+  def self.create_for(ontology_version)
+    # Although unlikely, the token could clash with another one. Generate tokens
+    # until there is no conflict. This should converge extremely fast.
+    access_token = nil
+    while access_token.nil? || access_token.invalid?
+      access_token = build_for(ontology_version)
+    end
+    access_token.save!
+    AccessTokenDeletionWorker.perform_at(access_token.expiration)
+    access_token
+  end
+
+  def self.fresh_expiration_date
+    Settings.access_token.expiration_minutes.minutes.from_now
+  end
+
+  def self.build_for(ontology_version)
+    repository = ontology_version.repository
+    AccessToken.new({
+      repository: repository,
+      expiration: fresh_expiration_date,
+      token: generate_token_string(ontology_version, repository)},
+      {without_protection: true})
```

Comment:
Align the parameters of a method call if they span more than one line.<br>Redundant curly braces around a hash parameter.<br>Space inside { missing.<br>Space inside } missing.